app/src/main/kotlin/com/simplemobiletools/smsmessenger/adapters/BaseConversationsAdapter.kt

```diff
@@ -0,0 +1,183 @@
+package com.simplemobiletools.smsmessenger.adapters
+
+import android.graphics.Typeface
+import android.os.Parcelable
+import android.util.TypedValue
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+import com.qtalk.recyclerviewfastscroller.RecyclerViewFastScroller
+import com.simplemobiletools.commons.adapters.MyRecyclerViewListAdapter
+import com.simplemobiletools.commons.extensions.*
+import com.simplemobiletools.commons.helpers.SimpleContactsHelper
+import com.simplemobiletools.commons.helpers.ensureBackgroundThread
+import com.simplemobiletools.commons.views.MyRecyclerView
+import com.simplemobiletools.smsmessenger.R
+import com.simplemobiletools.smsmessenger.activities.SimpleActivity
+import com.simplemobiletools.smsmessenger.extensions.*
+import com.simplemobiletools.smsmessenger.models.Conversation
+import kotlinx.android.synthetic.main.item_conversation.view.*
+
+@Suppress("LeakingThis")
+abstract class BaseConversationsAdapter(
+    activity: SimpleActivity, recyclerView: MyRecyclerView, onRefresh: () -> Unit, itemClick: (Any) -> Unit
+) : MyRecyclerViewListAdapter<Conversation>(activity, recyclerView, ConversationDiffCallback(), itemClick, onRefresh),
+    RecyclerViewFastScroller.OnPopupTextUpdate {
+    private var fontSize = activity.getTextSize()
+    private var drafts = HashMap<Long, String?>()
+
+    private var recyclerViewState: Parcelable? = null
+
+    init {
+        setupDragListener(true)
+        ensureBackgroundThread {
```

Comment:
this bg thread should be moved into fetchDrafts too, it just doesnt feel right here. Check if it works properly with updateDrafts too

Reply:
This was copied from existing code (`ConversationsAdapter` on master). Looking at the fetch drafts and the way it works, it would not be straightforward to make this change, since it would break `updateDrafts`. Maybe we can handle it in a separate PR, since I think it would require a bit more work (currently `fetchDrafts` fills in values by mutating the passed map, but we need these values right away in `updateDrafts`).

Reply:
okay, it is just a styling thing, no need to work on it too much, lets keep it as it is then